packages/toml/toml.7.0.0/opam

```diff
@@ -0,0 +1,49 @@
+opam-version: "2.0"
+synopsis: "Library for TOML with a parser, a serializer and a printer"
+description:
+  "toml is an OCaml library providing a parser, a serializer and a printer for TOML, a minimal configuration file format. Helpful getters to retrieve data as OCaml primitive types are also supplied."
+maintainer: ["OCamlPro <contact@ocamlpro.com>"]
+authors: [
+  "Julien Sagot"
+  "Emmanuel Surleau"
+  "mackwic"
+  "Andrew Rudenko"
+  "orbifx"
+  "c-cube"
+]
+license: "LGPL3"
+homepage: "https://ocaml-toml.github.io/To.ml/"
+doc: "https://ocaml-toml.github.io/To.ml/api/"
+bug-reports: "https://github.com/ocaml-toml/To.ml/issues"
+depends: [
+  "ocaml" {>= "4.07"}
+  "dune" {>= "2.7" & >= "2.7"}
+  "menhir" {build}
+  "ounit2" {with-test}
+  "bisect_ppx" {with-test & >= "2.5" & dev}
+  "ocb" {with-test & >= "0.1" & dev}
+  "odoc" {with-doc}
+  "ISO8601" {>= "0.2"}
+]
+build: [
+  ["dune" "subst"] {dev}
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
```

Comment:
```suggestion
   #  "@runtest" {with-test}
```

Reply:
They should work for users. If you look closely at opam-ci results, they pass almost everywhere. They fail on debian testing and unstable, I'm using unstable myself and they pass for me. I guess it's only because of the CI system being configured in a special way.

But I don't care to disable them. So,if you think it's the better option, feel free to do it. :-)

Reply:
The tests fail on all OCaml compiler versions from 4.07 to 4.12, not just on debian-testing/unstable. It's too long for me to spot a pattern in the diffs, though.

```
2021-03-16 16:47.23: >>> Fatal error: exception File "tests/lenses_test.ml", line 282, characters 4-10: Assertion failed (score = 20)
2021-03-16 16:47.23: >>> Fatal error: exception OUnitTest.OUnit_failure("expected: 1979-05-27T07:32:00+00:00 but got: 1979-05-27T08:32:00+00:00") (score = 20)
2021-03-16 16:47.23: >>> Fatal error: exception OUnitTest.OUnit_failure("expected: title = \"TOML Example\"\n[clients]\ndata = [[\"gamma\", \"delta\"], [1, 2]]\nhosts = [\"alpha\", \"omega\"]\n[database]\nconnection_max = 5000\nenabled = true\nports = [8001, 8001, 8002]\nserver = \"192.168.1.1\"\n[owner]\nbio = \"\"\"GitHub Cofounder & CEO\nLikes tater tots and beer.\"\"\"\ndob = 1979-05-27T08:32:00+00:00\nname = \"Tom Preston-Werner\"\norganization = \"GitHub\"\n[[products]]\nname = \"Hammer\"\nsku = 738594937\n[[products]]\ncolor = \"gray\"\nname = \"Nail\"\nsku = 284758393\n[servers.alpha]\ndc = \"eqdc10\"\nip = \"10.0.0.1\"\n[servers.beta]\ncountry = \"\228\184\173\229\155\189\"\ndc = \"eqdc10\"\nip = \"10.0.0.2\"\n\nbut got: title = \"TOML Example\"\n[clients]\ndata = [[\"gamma\", \"delta\"], [1, 2]]\nhosts = [\"alpha\", \"omega\"]\n[database]\nconnection_max = 5000\nenabled = true\nports = [8001, 8001, 8002]\nserver = \"192.168.1.1\"\n[owner]\nbio = \"\"\"GitHub Cofounder & CEO\nLikes tater tots and beer.\"\"\"\ndob = 1979-05-27T07:32:00+00:00\nname = \"Tom Preston-Werner\"\norganization = \"GitHub\"\n[[products]]\nname = \"Hammer\"\nsku = 738594937\n[[products]]\ncolor = \"gray\"\nname = \"Nail\"\nsku = 284758393\n[servers.alpha]\ndc = \"eqdc10\"\nip = \"10.0.0.1\"\n[servers.beta]\ncountry = \"\228\184\173\229\155\189\"\ndc = \"eqdc10\"\nip = \"10.0.0.2\"\n") (score = 20)
2021-03-16 16:47.23: >>> Fatal error: exception OUnitTest.OUnit_failure("expected: [array]\nkey1 = [1, 2, 3]\nkey2 = [\"red\", \"yellow\", \"green\"]\nkey3 = [[1, 2], [3, 4, 5]]\nkey4 = [[1, 2], [\"a\", \"b\", \"c\"]]\nkey5 = [1, 2, 3]\nkey6 = [1, 2]\n[[array.inline.points]]\nx = 1\ny = 2\nz = 3\n[[array.inline.points]]\nx = 7\ny = 8\nz = 9\n[[array.inline.points]]\nx = 2\ny = 4\nz = 8\n[boolean]\nFalse = false\nTrue = true\n[datetime]\nkey1 = 1979-05-27T01:00:00+00:00\nkey2 = 1979-05-27T08:32:00+00:00\nkey3 = 1979-05-27T08:32:00+00:00\nkey4 = 1979-05-27T08:32:00+00:00\n[float.both]\nkey = 0.0\n[float.exponent]\nkey1 = 49999999999999995805696.0\nkey2 = 1000000.0\nkey3 = -0.02\n[float.fractional]\nkey1 = 1.0\nkey2 = 3.1415\nkey3 = -0.01\n[float.underscores]\nkey1 = 9224617.44599\nkey2 = inf\n[[fruit]]\nname = \"apple\"\n[fruit.physical]\ncolor = \"red\"\nshape = \"round\"\n[[fruit.variety]]\nname = \"red delicious\"\n[[fruit.variety]]\nname = \"granny smith\"\n[[fruit]]\nname = \"banana\"\n[[fruit.variety]]\nname = \"plantain\"\n[integer]\nkey1 = 99\nkey2 = 42\nkey3 = 0\nkey4 = -17\n[integer.underscores]\nkey1 = 1000\nkey2 = 5349221\nkey3 = 12345\n[[products]]\nname = \"Hammer\"\nsku = 738594937\n[[products]]\ncolor = \"gray\"\nname = \"Nail\"\nsku = 284758393\n[string.basic]\nbasic = \"I'm a string. \\\"You can quote me\\\". Name\\tJos\195\169\\nLocation\\tSF.\"\n[string.literal]\nquoted = \"Tom \\\"Dubs\\\" Preston-Werner\"\nregex = \"<\\\\i\\\\c*\\\\s*>\"\nwinpath = \"C:\\\\Users\\\\nodejs\\\\templates\"\nwinpath2 = \"\\\\\\\\ServerX\\\\admin$\\\\system32\\\\\"\n[string.literal.multiline]\nlines = \"\"\"The first newline is\ntrimmed in raw strings.\n   All other whitespace\n   is preserved.\n\"\"\"\nregex2 = \"I [dw]on't need \\\\d{2} apples\"\n[string.multiline]\nkey1 = \"\"\"One\nTwo\"\"\"\nkey2 = \"\"\"One\nTwo\"\"\"\nkey3 = \"\"\"One\nTwo\"\"\"\n[string.multiline.continued]\nkey1 = \"The quick brown fox jumps over the lazy dog.\"\nkey2 = \"The quick brown fox jumps over the lazy dog.\"\nkey3 = \"The quick brown fox jumps over the lazy dog.\"\n[table]\nkey = \"value\"\n[table.inline.name]\nfirst = \"Tom\"\nlast = \"Preston-Werner\"\n[table.inline.point]\nx = 1\ny = 2\n[table.subtable]\nkey = \"another value\"\n\nbut got: [array]\nkey1 = [1, 2, 3]\nkey2 = [\"red\", \"yellow\", \"green\"]\nkey3 = [[1, 2], [3, 4, 5]]\nkey4 = [[1, 2], [\"a\", \"b\", \"c\"]]\nkey5 = [1, 2, 3]\nkey6 = [1, 2]\n[[array.inline.points]]\nx = 1\ny = 2\nz = 3\n[[array.inline.points]]\nx = 7\ny = 8\nz = 9\n[[array.inline.points]]\nx = 2\ny = 4\nz = 8\n[boolean]\nFalse = false\nTrue = true\n[datetime]\nkey1 = 1979-05-27T00:00:00+00:00\nkey2 = 1979-05-27T07:32:00+00:00\nkey3 = 1979-05-27T07:32:00+00:00\nkey4 = 1979-05-27T07:32:00+00:00\n[float.both]\nkey = 0.0\n[float.exponent]\nkey1 = 49999999999999995805696.0\nkey2 = 1000000.0\nkey3 = -0.02\n[float.fractional]\nkey1 = 1.0\nkey2 = 3.1415\nkey3 = -0.01\n[float.underscores]\nkey1 = 9224617.44599\nkey2 = inf\n[[fruit]]\nname = \"apple\"\n[fruit.physical]\ncolor = \"red\"\nshape = \"round\"\n[[fruit.variety]]\nname = \"red delicious\"\n[[fruit.variety]]\nname = \"granny smith\"\n[[fruit]]\nname = \"banana\"\n[[fruit.variety]]\nname = \"plantain\"\n[integer]\nkey1 = 99\nkey2 = 42\nkey3 = 0\nkey4 = -17\n[integer.underscores]\nkey1 = 1000\nkey2 = 5349221\nkey3 = 12345\n[[products]]\nname = \"Hammer\"\nsku = 738594937\n[[products]]\ncolor = \"gray\"\nname = \"Nail\"\nsku = 284758393\n[string.basic]\nbasic = \"I'm a string. \\\"You can quote me\\\". Name\\tJos\195\169\\nLocation\\tSF.\"\n[string.literal]\nquoted = \"Tom \\\"Dubs\\\" Preston-Werner\"\nregex = \"<\\\\i\\\\c*\\\\s*>\"\nwinpath = \"C:\\\\Users\\\\nodejs\\\\templates\"\nwinpath2 = \"\\\\\\\\ServerX\\\\admin$\\\\system32\\\\\"\n[string.literal.multiline]\nlines = \"\"\"The first newline is\ntrimmed in raw strings.\n   All other whitespace\n   is preserved.\n\"\"\"\nregex2 = \"I [dw]on't need \\\\d{2} apples\"\n[string.multiline]\nkey1 = \"\"\"One\nTwo\"\"\"\nkey2 = \"\"\"One\nTwo\"\"\"\nkey3 = \"\"\"One\nTwo\"\"\"\n[string.multiline.continued]\nkey1 = \"The quick brown fox jumps over the lazy dog.\"\nkey2 = \"The quick brown fox jumps over the lazy dog.\"\nkey3 = \"The quick brown fox jumps over the lazy dog.\"\n[table]\nkey = \"value\"\n[table.inline.name]\nfirst = \"Tom\"\nlast = \"Preston-Werner\"\n[table.inline.point]\nx = 1\ny = 2\n[table.subtable]\nkey = \"another value\"\n") (score = 20)
```

Reply:
They are like daysavings failures (1h difference in some tests), I did have a look last time we had asked

Reply:
@zapashcanon may be worth trying to replicate the issues locally using `ocaml/opam:debian-10-ocaml-4.*` or the corresponding `debian-unstable` images and see if there is anything that may need fixing in a new release (or why the images are doing something wrong).

Reply:
> The tests fail on all OCaml compiler versions from 4.07 to 4.12,

I bet they are running on debian testing/unstable, right ?

Reply:
Yes, that is why I was suggesting to use those two images